modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java

```diff
@@ -985,6 +989,76 @@ public void testAddObjectAction() throws Exception {
 		_objectActionLocalService.deleteObjectAction(systemObjectAction);
 	}
 
+	@Test
+	public void testAddObjectActionInSystemObjectDefinitionWithConditionExpression()
+		throws Exception {
+
+		SystemObjectDefinitionManager userSystemObjectDefinitionManager =
+			_systemObjectDefinitionManagerRegistry.
+				getSystemObjectDefinitionManager("User");
+
+		ObjectDefinition userObjectDefinition =
+			_objectDefinitionLocalService.fetchSystemObjectDefinition(
+				TestPropsValues.getCompanyId(), "User");
+
+		ObjectFieldTestUtil.addCustomObjectField(
+			TestPropsValues.getUserId(),
+			ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+			ObjectFieldConstants.DB_TYPE_STRING, userObjectDefinition,
+			"userCustomField");
+
+		NotificationTemplate notificationTemplate =
+			_notificationTemplateLocalService.addNotificationTemplate(
+				RandomTestUtil.randomString(),
+				UserTestUtil.getAdminUser(
+					TestPropsValues.getCompanyId()
+				).getUserId(),
+				NotificationConstants.TYPE_EMAIL);
+
+		String expectedValue = RandomTestUtil.randomString();
+
+		_objectActionLocalService.addObjectAction(
+			RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+			userObjectDefinition.getObjectDefinitionId(), true,
+			"oldValue(\"userCustomField\") == \"" + expectedValue + "\"",
+			RandomTestUtil.randomString(),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			RandomTestUtil.randomString(),
+			ObjectActionExecutorConstants.KEY_NOTIFICATION,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_UPDATE,
+			UnicodePropertiesBuilder.put(
+				"notificationTemplateId",
+				notificationTemplate.getNotificationTemplateId()
+			).build(),
+			false);
+
+		JSONObject userAccountJSONObject =
+			UserAccountTestUtil.addUserAccountJSONObject(
+				userSystemObjectDefinitionManager,
+				HashMapBuilder.<String, Serializable>put(
+					"userCustomField", expectedValue
+				).build());
```

Comment:
@JhossephG you can simplify by using the `systemObjectDefinitionManager.addBaseModel` directly here.

```
long userId = systemObjectDefinitionManager.addBaseModel(
	TestPropsValues.getUser(),
	HashMapBuilder.<String, Object>put(
		"alternateName", RandomTestUtil.randomString()
	).put(
		"emailAddress", RandomTestUtil.randomString() + "@liferay.com"
	).put(
		"familyName", RandomTestUtil.randomString()
	).put(
		"givenName", RandomTestUtil.randomString()
	).put(
		"name", "Paul"
	).build());
```

This way you don't need to import a new package.